### PR TITLE
!chore: write upgrade handler for new minimum commission rate

### LINKED ISSUE #151: Chore: Upgrade handler to modify commission %
With the passing of prop 8, the network is now enforcing a min commission percentage of 5%. To wrap up the migration and prop, we need to update the existing validators all to 5%. Upon speaking with the validators we have agreed this is the best path forward.

### PR DESCRIPTION
**Pull request by neitdung**
closes #151

---

Description:
- Change staking params' min commission rate to 5%
- Update all validators commission rate

### CODE REVIEW

**Comment by catShaark** [comment]
can you write some tests for this

File: app/upgrades/v4/constants.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package v4
+
+const (
+	// UpgradeName defines the on-chain upgrade name.
+	UpgradeName         = "v4-Commission"
```
Comment (catShaark) [inline app/upgrades/v4/constants.go:5]: This should be coupled with sdk47 upgrade

File: app/upgrades/v4/upgrades.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+func fixMinCommisionRate(ctx sdk.Context, staking stakingkeeper.Keeper) {
+	// Upgrade every validators min-commission rate
+	validators := staking.GetAllValidators(ctx)
+	newComm := sdk.MustNewDecFromStr(NewMinCommisionRate)
+	params := staking.GetParams(ctx)
+	params.MinCommissionRate = newComm
```
Comment (catShaark) [inline app/upgrades/v4/upgrades.go]: newComm -> minComm

File: app/upgrades/v4/constants.go (hunk 1 above)
Comment (neitdung) [reply app/upgrades/v4/constants.go:5]: yes, we'll change in sdk47 PR

File: app/upgrades/v4/upgrades.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+			comm, err := updateValidatorCommission(ctx, staking, v, newComm)
+			if err != nil {
+				panic(err)
+			}
+
+			v.Commission = comm
+
+			// call the before-modification hook since we're about to update the commission
+			staking.BeforeValidatorModified(ctx, v.GetOperator())
+			staking.SetValidator(ctx, v)
```
Comment (catShaark) [inline app/upgrades/v4/upgrades.go:49]: is there staking.AfterValidatorModified() ?

Comment (neitdung) [reply app/upgrades/v4/upgrades.go:49]: No, it just have AfterValidatorCreated, AfterValidatorRemoved, AfterValidatorBonded and AfterValidatorBeginUnbonding

File: app/upgrades_test.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	app.StakingKeeper.SetValidator(ctx, validators[0])
+	app.StakingKeeper.SetValidator(ctx, validators[1])
+	app.StakingKeeper.SetValidator(ctx, validators[2])
+	app.StakingKeeper.SetValidator(ctx, validators[3])
+
+	upgradePlan := upgradetypes.Plan{
+		Name:   v4.UpgradeName,
+		Height: ctx.BlockHeight(),
+	}
+	err := app.UpgradeKeeper.ScheduleUpgrade(ctx, upgradePlan)
```
Comment (catShaark) [inline app/upgrades_test.go:57]: can you have some vals with above minComm commission rate, also set staking min comm rate to be the `NewMinCommisionRate`

**Comment by catShaark** [comment]
I think this is good, tho this should target sdk47 branch

**Comment by neitdung** [comment]
Closed by [realio-network](https://github.com/decentrio/realio-network/pull/8)

**Pull request closed without merging**